apps/src/code-studio/pd/form_components/FormController.jsx

```diff
@@ -269,7 +268,9 @@ export default class FormController extends React.Component {
    * @returns {String[]}
    */
   getRequiredFields() {
-    return this.props.requiredFields;
+    const requiredFields = [...this.props.requiredFields];
+    this.getPageComponents().forEach(page => requiredFields.push(...page.getDynamicallyRequiredFields(this.state.data)));
```

Comment:
why not `map` then `concat`?

Reply:
Because I didn't think of it :)
I like it! Will update.